src/Crowdin.Api/CrowdinApiClient.cs

```diff
@@ -308,14 +323,23 @@ public Task<CrowdinApiResult> UploadFile(string subUrl, string filename, Stream
             return outResultList.ToArray();
         }
 
-        private async Task<CrowdinApiResult> SendRequest(HttpRequestMessage request)
+        private HttpClient GetHttpClientInstance()
+        {
+            var httpClient = _httpClientFactory();
+            httpClient.DefaultRequestHeaders.Authorization =
+                new AuthenticationHeaderValue("Bearer", _accessToken);
+
+            return httpClient;
+        }
+
+        private async Task<CrowdinApiResult> SendRequest(Func<HttpRequestMessage> requestFn)
         {
             var result = new CrowdinApiResult();
-            
+
             HttpResponseMessage response =
                 _retryService != null
-                ? await _retryService.ExecuteRequestAsync(() => _httpClient.SendAsync(request))
-                : await _httpClient.SendAsync(request);
+                ? await _retryService.ExecuteRequestAsync(() => GetHttpClientInstance().SendAsync(requestFn()))
```

Comment:
Creating a new instance of `HttpClient` per each request is not recommended approach by Microsoft and can lead to connection pool exhaustion. Sources : [one](https://learn.microsoft.com/en-us/aspnet/web-api/overview/advanced/calling-a-web-api-from-a-net-client#create-and-initialize-httpclient), [two](https://learn.microsoft.com/en-us/azure/architecture/antipatterns/improper-instantiation/).

I suggest mix the approaches. We can use the same instance of `HttpClient`, but create `HttpRequestMessage`-s per each new request rather than reusing the same message (because it's disposable). I tested this approach and it works well for me. Could you please also check it?

Here is a snippet :

```csharp
private async Task<CrowdinApiResult> SendRequest(Func<HttpRequestMessage> requestFn) {

HttpResponseMessage response = _retryService != null
  ? await _retryService.ExecuteRequestAsync(() => _httpClient.SendAsync(requestFn()))
  : await _httpClient.SendAsync(requestFn());

// Continuation
```

Reply:
Thank you @innomaxx .

I totally agree. Verified on my end and initiating only the `HttpRequestMessage` per request does the job.
New changes were pushed.

Please advise on the following:
1. What should I do about the test coverage? It's not really possible to unit tests those changes.
2. Should I manually update the nuspec and README files within this PR?

Reply:
Thank you @PatrykPlewaOfficial.

1) I think there is nothing to do here now. We at Crowdin will take into consideration adding tests also for the request dispatcher and related mechanisms of this library.

2) Since the library's public API is not changed - I don't think extra changes needed for the examples in README. Also we usually update nuspec and related meta files with each release.

Thank you again for your contribution!